packages/contracts-create2factory/src/buildinfo.ts

```diff
@@ -0,0 +1,125 @@
+import { Address } from "viem";
+import { readFileSync } from "fs";
+import { BuildInfo, HardhatMetadata, StandardJSONInput } from "./types/buildinfo.js";
+import { etherscanGetAbi, etherscanVerifySourceCode, waitForEtherscanVerifyStatus } from "./utils/etherscanUtils.js";
+
+// From hardhat-core
+/**
+ * A BuildInfo is a file that contains all the information of a solc run. It
+ * includes all the necessary information to recreate that exact same run, and
+ * all of its output.
+ */
+
+async function verifyContract({
+    apiUrl,
+    apiKey,
+    contractAddress,
+    contractName,
+    metadata,
+    compilerVersion,
+}: {
+    apiUrl: string;
+    apiKey: string;
+    contractAddress: Address;
+    contractName: string;
+    metadata: HardhatMetadata;
+    compilerVersion: string;
+}) {
+    const abiCheck = await etherscanGetAbi({ apiUrl, apiKey, contractAddress });
+
+    if (abiCheck) {
+        console.log("Contract already verified.");
```

Comment:
Maybe comment out the logging or make it more specific using `contractName` and `contractAddress` parameters.